python/nano/src/bigdl/nano/deps/neural_compressor/pytorch/quantized_model.py

```diff
@@ -0,0 +1,51 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+
+import yaml
+from bigdl.nano.utils.inference.pytorch.model import AcceleratedLightningModule
+from neural_compressor.utils.pytorch import load
+from neural_compressor.model.model import PyTorchModel
+
+
+class QuantizedModel(AcceleratedLightningModule):
+    def __init__(self, model):
+        super().__init__(model.model)
+        self.quantized = model
+
+    def save(self, path):
+        return self.quantized.save(path)
+
+    def load_state_dict(self, state_dict):
+        load(state_dict, self.quantized._model)
+
+    def state_dict(self):
+        try:
+            stat_dict = self.quantized.model.state_dict()
+            stat_dict['best_configure'] = self.quantized.tune_cfg
+        except IOError as e:
+            raise IOError("Fail to dump configure and weights due to {}.".format(e))
+        return stat_dict
+
+    @staticmethod
+    def load(path, model):
```

Comment:
it should be `load(path)` that matches `save(path)`

Reply:
For inc, we need a fp32 model to load quantized model. This calls `load` provided by inc.
https://github.com/intel/neural-compressor/blob/v1.8.1/neural_compressor/utils/pytorch.py#L68-L81

Reply:
We should handle this automatically, instead of having the user to handle it.

Reply:
Yes, my intention for this function was to help uses to quickly deploy saved model on another machine.  
For openvino, users can simply do `PytorchOpenVINOModel.load(path)` to create a model for inference.  
Different from openvino, pytorch saves weights only. So in traditional usage, we usually first construction the model graph and then load weights.  
I need to investigate a bit more about this saving function.

Reply:
How doe the save-load workflow look like in these cases?

- For ONNX or OpenVINO???
  ```
  qmodel = trainer.quantize(model, accelerator="ONNXRT", ...)
  qmode.save(path)
  qmodel = model.load(path)
  ```

 - For PyTorch???
   ```
   qmodel = trainer.quantize(model, accelerator="ONNXRT", ...)
   qmode.save(path)
   ???
   ```